packages/vuetify/src/composables/dateFormat.ts

```diff
@@ -0,0 +1,131 @@
+// Composables
+import { useDate } from '@/composables/date/date'
+
+// Utilities
+import { toRef } from 'vue'
+import { propsFactory } from '@/util'
+
+// Types
+import type { Ref } from 'vue'
+
+// Types
+export interface DateFormatProps {
+  inputFormat?: string
+}
+
+class DateFormatSpec {
+  constructor (
+    public readonly order: string, // mdy | dmy | ymd
+    public readonly separator: string // / | - | .
+  ) { }
+
+  get format () {
+    return this.order.split('')
+      .map(sign => `${sign}${sign}`)
+      .join(this.separator)
+      .replace('yy', 'yyyy')
+  }
+
+  static canBeParsed (v: any) {
+    if (typeof v !== 'string') return false
+    const lowercase = v.toLowerCase()
+    return ['y', 'm', 'd'].every(sign => lowercase.includes(sign)) &&
+      ['/', '-', '.'].some(sign => v.includes(sign))
+  }
+
+  static parse (v: string) {
+    if (!DateFormatSpec.canBeParsed(v)) {
+      throw new Error(`[${v}] cannot be parsed into date format specification`)
+    }
+    const order = v.toLowerCase().split('')
+      .filter((c, i, all) => 'dmy'.includes(c) && all.indexOf(c) === i)
+      .join('')
+    const separator = ['/', '-', '.'].find(sign => v.includes(sign))!
+    return new DateFormatSpec(order, separator)
+  }
+}
+
+export const makeDateFormatProps = propsFactory({
+  inputFormat: {
+    type: String,
+    validator: (v: string) => !v || DateFormatSpec.canBeParsed(v),
+  },
+}, 'lazy')
+
+export function useDateFormat (props: DateFormatProps, locale: Ref<string>) {
+  const adapter = useDate()
+
+  function inferFromLocale () {
+    return Intl.DateTimeFormat(locale.value ?? 'en-US', { year: 'numeric', month: '2-digit', day: '2-digit' })
+      .format(new Date(1999, 11, 7))
```

Comment:
How about:
```js
.format(adapter.parseISO('1999-11-07'))
```

Want to avoid using JS Date as much as possible and reuse our adapter so that when we switch to Temporal, the transition would be smooth

Reply:
You only get `Date` if you use default adapter. `Intl` only works with native `Date`.
It will need to be `adapter.toJsDate(adapter.parseISO('1999-11-07'))`.

Good catch! I get different results when simulating time zone from left hemisphere.

```js
.format(adapter.toJsDate(adapter.parseISO('1999-11-07'))) // --> 11/07/1999
.format(new Date('1999-11-07')) // --> 11/06/1999
```